app/models/concerns/status/visibility.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Status::Visibility
+  extend ActiveSupport::Concern
+
+  DISTRIBUTABLE_VISIBILITIES = %i(public unlisted).freeze
+  LIST_ELIGIBLE_VISIBILITIES = %i(public unlisted private).freeze
+  RESTRICTED_VISIBILITIES = %w(direct limited).freeze
+
+  included do
+    enum :visibility,
+         { public: 0, unlisted: 1, private: 2, direct: 3, limited: 4 },
+         suffix: :visibility,
+         validate: true
+
+    scope :distributable_visibility, -> { where(visibility: DISTRIBUTABLE_VISIBILITIES) }
```

Comment:
It would be INTERESTING to me if, at the Rails level, you could have like a "named subgroups" feature and do something like:

```ruby
enum :visibility,
  { public: 0, unlisted: 1, private: 2 },
  groups: { fun: %i[public private] },
  suffix: :visibility
```

and that would generate all the normal scopes/query methods, but then also generate `.fun_visibility` scope and `fun_visibility?` query method, which would wrap that collection of values.